proxy/test/proxy.test.ts

```diff
@@ -0,0 +1,163 @@
+import { equal } from 'node:assert'
+import { createServer } from 'node:http'
+import { before, describe, test } from 'node:test'
+import { URL } from 'node:url'
+
+import createProxyServer from '../proxy.js'
+import { getTestHttpServer, initTestHttpServer } from './utils.js'
+
+/**
+ * Proxy is tested with a service that outputs request and response data.
+ *
+ * <local client> -> <proxy> -> <targetServer>
+ */
+const targetServer = createServer((req, res) => {
+  let parsedUrl = new URL(req.url || '', `http://${req.headers.host}`)
+
+  let queryParams = Object.fromEntries(parsedUrl.searchParams.entries())
+  let requestPath = parsedUrl.pathname
+
+  res.writeHead(200, {
+    'Content-Type': 'text/json'
+  })
+
+  res.end(
+    JSON.stringify({
+      request: {
+        headers: req.headers,
+        method: req.method,
+        queryParams,
+        requestPath,
+        url: req.url
+      },
+      response: 'ok'
+    })
+  )
+})
+
+describe('proxy tests', async () => {
+  await initTestHttpServer(
+    'proxy',
+    createProxyServer({ hostnameWhitelist: ['localhost'], silentMode: true })
+  )
+  await initTestHttpServer('target', targetServer)
+
+  let proxyServerUrl = ''
+  let targetServerUrl = ''
+
+  before(() => {
+    let proxy = getTestHttpServer('proxy')
+    if (proxy) {
+      proxyServerUrl = proxy.baseUrl
+    }
+
+    let target = getTestHttpServer('target')
+    if (target) {
+      targetServerUrl = target.baseUrl
+    }
+
+    if (!target || !proxy) {
+      throw new Error(
+        "Couldn't set up target server or proxy server. Something is misconfigured. Please check out 'proxy.test.js'"
```

Comment:
Can we make it shorter to fit `80` columns line?

Reply:
Just made that comment shorter! Thanks